jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -364,6 +364,14 @@ re_match_regexp (re_matcher_ctx_t *re_ctx_p, /**< RegExp matcher context */
                  const lit_utf8_byte_t *str_p, /**< input string pointer */
                  const lit_utf8_byte_t **out_str_p) /**< [out] matching substring iterator */
 {
+#ifdef REGEXP_RECURSION_LIMIT
+  if (re_ctx_p->recursion_depth >= REGEXP_RECURSION_LIMIT)
+  {
+    ecma_value_t ret_value = ecma_raise_range_error ("RegExp executor recursion limit is exceeded.");
+    return ret_value;
```

Comment:
You don't need to create a local variable for this.

Reply:
You are right, thank you for the review! I've updated the PR with this change.